homeassistant/components/nexia/diagnostics.py

```diff
@@ -0,0 +1,32 @@
+"""Diagnostics support for nexia."""
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant.components.diagnostics import async_redact_data
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+
+from .const import CONF_BRAND, DOMAIN
+from .coordinator import NexiaDataUpdateCoordinator
+
+TO_REDACT = {
+    "dealer_contact_info",
+}
+
+
+async def async_get_config_entry_diagnostics(
+    hass: HomeAssistant, entry: ConfigEntry
+) -> dict[str, Any]:
+    """Return diagnostics for a config entry."""
+    coordinator: NexiaDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+    nexia_home = coordinator.nexia_home
+
+    return {
+        "entry": {
+            "title": entry.title,
+            "brand": entry.data.get(CONF_BRAND),
```

Comment:
Any specific reason to not drop the whole entry in here? (with username/password, house_id redacted)

Reply:
house_id seems to be part of the links data anyway as well. Might be sensitive?

Reply:
It is not sensitive. It is a randomly assigned identifier.

I didn't add it to the data since the only thing we really ever have problems with is the brand

Reply:
Randomly assigned identifiers might be considered secret (webhooks and stuff). Anyway i think this looks good, even though that HUGE test data is a bit ugly.

Reply:
All of them require authentication to access so I think its OK.  We would need to consider the zone ids, thermostat ids, automation ids as sensitive if we went that route so I think that wouldn't be tenable.